newIDE/app/src/PlatformSpecificAssetsEditor/PlatformSpecificAssets.js

```diff
@@ -0,0 +1,308 @@
+        })),
+        (async () => ({
+          resourceName: 'android-windowSplashScreenAnimatedIcon.png',
+          blobDataUrl: await resizeImage(chosenFileAsBlobDataUrl, {
+            width: androidWindowSplashScreenAnimatedIconRecommendedSize,
+            height: androidWindowSplashScreenAnimatedIconRecommendedSize,
+            transparentBorderSize:
+              androidWindowSplashScreenAnimatedIconRecommendedSize / 6,
+          }),
+        }))(),
+        ...iosIconSizes.map(async size => ({
+          resourceName: `ios-icon-${size}.png`,
+          blobDataUrl: await resizeImage(chosenFileAsBlobDataUrl, {
+            width: size,
+            height: size,
+          }),
+        })),
+      ]);
+
+      results.forEach(({ resourceName, blobDataUrl }) => {
+        const metadata = JSON.stringify({
+          extension: '.png',
+          // Used in local app to define the path where to save the image on
+          // the file system.
+          localFilePath: `./assets/${resourceName}`,
+        });
+        if (!resourcesManager.hasResource(resourceName)) {
+          const imageResource = new gd.ImageResource();
+          imageResource.setFile(blobDataUrl);
+          imageResource.setName(resourceName);
+          imageResource.setMetadata(metadata);
+
+          resourcesManager.addResource(imageResource);
+
+          // Important, we are responsible for deleting the resources that we created
+          // Otherwise we have a memory leak, as calling addResource is making a copy of the resource.
+          imageResource.delete();
+        } else {
+          resourcesManager.getResource(resourceName).setFile(blobDataUrl);
+          resourcesManager.getResource(resourceName).setMetadata(metadata);
+        }
+      });
+
+      await resourceManagementProps.onFetchNewlyAddedResources();
+
+      // Make sure the resources are (re)loaded.
+      ResourcesLoader.burstUrlsCacheForResources(
+        project,
+        results.map(({ resourceName }) => resourceName)
+      );
+      setTimeout(() => {
+        onDesktopIconResourceNamesChanged(
+          desktopIconSizes.map(size => `desktop-icon-${size}.png`)
+        );
+        onAndroidIconResourceNamesChanged(
+          androidIconSizes.map(size => `android-icon-${size}.png`)
+        );
+        onAndroidWindowSplashScreenAnimatedIconResourceNameChanged(
+          'android-windowSplashScreenAnimatedIcon.png'
+        );
+        onIosIconResourceNamesChanged(
+          iosIconSizes.map(size => `ios-icon-${size}.png`)
+        );
+        setIsLoading(false);
+      }, 200 /* Let a bit of time so that image files can be found */);
+    } catch (e) {
+      showAlert({
+        title: t`Some icons could not be generated.`,
+        message: t`An error occurred while generating some icons. Verify that the image is valid and try again.`,
+      });
+      setIsLoading(false);
+    } finally {
+      // Reset input value so that if the user selects the same file again,
```

Comment:
See if you can/should call [revokeObjectURL()](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static) here?